### ISSUE TITLE
"null" as a topic in filter parameters does not function as a wildcard 

### ISSUE THREAD

**Issue by zwx00** [ethereum/go-ethereum#15187]
#### System information

Geth version: 

Geth
Version: 1.7.0-unstable
Git Commit: 382c9266e6f30e3a6cdcb9ec08ffa51dacb71c87
Architecture: amd64
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.9
Operating System: linux
GOPATH=
GOROOT=/usr/lib/go

OS & Version: 
Linux ** 4.9.50-1-MANJARO #1 SMP PREEMPT Thu Sep 14 09:07:15 UTC 2017 x86_64 GNU/Linux

#### Expected behaviour

```
> eth.filter({'fromBlock': 0, 'toBlock': 'latest', address: '0xe25b529f5b2475694eeb29dced46dde986488b03', topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059", null, null]}).get().length
6125
> eth.filter({'fromBlock': 0, 'toBlock': 'latest', address: '0xe25b529f5b2475694eeb29dced46dde986488b03', topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059"]}).get().length
14286
```

These should both return the same value. Same issue is concerning in situations when I want to filter by the second topic, leaving the first one blank, as seen below.

#### Actual behaviour

```
> eth.filter({'fromBlock': 'earliest', 'toBlock': 'latest', address: '0xe25b529f5b2475694eeb29dced46dde986488b03', topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059", "0x74696d656c6f7264000000000000000000000000000000000000000000000000", "0x736f6d656f6e6520233400000000000000000000000000000000000000000000"]}).get()
[{
    address: "0xe25b529f5b2475694eeb29dced46dde986488b03",
    blockHash: "0xa8847f8ac2878fea7adccd60cd41afb2f0aa84271588db429bde4ef2fd3b4e84",
    blockNumber: 24939,
    data: "0x000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000f0000000000000000000000000000000000000000000000000000000059c3a79d",
    logIndex: 3,
    removed: false,
    topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059", "0x74696d656c6f7264000000000000000000000000000000000000000000000000", "0x736f6d656f6e6520233400000000000000000000000000000000000000000000"],
    transactionHash: "0x60e2fbbd600356332f520f4034e1cb06522644c4866056ed0ecce98a2f65706d",
    transactionIndex: 3
}]

> eth.filter({'fromBlock': 'earliest', 'toBlock': 'latest', address: '0xe25b529f5b2475694eeb29dced46dde986488b03', topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059", null, "0x736f6d656f6e6520233400000000000000000000000000000000000000000000"]}).get()
[]

> eth.filter({'fromBlock': 'earliest', 'toBlock': 'latest', address: '0xe25b529f5b2475694eeb29dced46dde986488b03', topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059", "0x74696d656c6f7264000000000000000000000000000000000000000000000000"]}).get()
[{
    address: "0xe25b529f5b2475694eeb29dced46dde986488b03",
    blockHash: "0xa8847f8ac2878fea7adccd60cd41afb2f0aa84271588db429bde4ef2fd3b4e84",
    blockNumber: 24939,
    data: "0x000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000f0000000000000000000000000000000000000000000000000000000059c3a79d",
    logIndex: 0,
    removed: false,
    topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059", "0x74696d656c6f7264000000000000000000000000000000000000000000000000", "0x736f6d656f6e6520233100000000000000000000000000000000000000000000"],
    transactionHash: "0xc035e6efc2adbb690a0042ec1c2a1aa253d1a6ddd4ce21431c9bf1cc0f73cec6",
    transactionIndex: 0
}, {
    address: "0xe25b529f5b2475694eeb29dced46dde986488b03",
    blockHash: "0xa8847f8ac2878fea7adccd60cd41afb2f0aa84271588db429bde4ef2fd3b4e84",
    blockNumber: 24939,
    data: "0x000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000f0000000000000000000000000000000000000000000000000000000059c3a79d",
    logIndex: 1,
    removed: false,
    topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059", "0x74696d656c6f7264000000000000000000000000000000000000000000000000", "0x736f6d656f6e6520233200000000000000000000000000000000000000000000"],
    transactionHash: "0x42efe721d760389813b4d4ca830fe1ad4e42af9159b8e3e944596eacc99085c0",
    transactionIndex: 1
}, {
    address: "0xe25b529f5b2475694eeb29dced46dde986488b03",
    blockHash: "0xa8847f8ac2878fea7adccd60cd41afb2f0aa84271588db429bde4ef2fd3b4e84",
    blockNumber: 24939,
    data: "0x000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000f0000000000000000000000000000000000000000000000000000000059c3a79d",
    logIndex: 2,
    removed: false,
    topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059", "0x74696d656c6f7264000000000000000000000000000000000000000000000000", "0x736f6d656f6e6520233300000000000000000000000000000000000000000000"],
    transactionHash: "0xfadd91219c305143628075f1b7d9ea4704d24a6f4c504da0cbd0bef774bea5e7",
    transactionIndex: 2
}, {
    address: "0xe25b529f5b2475694eeb29dced46dde986488b03",
    blockHash: "0xa8847f8ac2878fea7adccd60cd41afb2f0aa84271588db429bde4ef2fd3b4e84",
    blockNumber: 24939,
    data: "0x000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000f0000000000000000000000000000000000000000000000000000000059c3a79d",
    logIndex: 3,
    removed: false,
    topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059", "0x74696d656c6f7264000000000000000000000000000000000000000000000000", "0x736f6d656f6e6520233400000000000000000000000000000000000000000000"],
    transactionHash: "0x60e2fbbd600356332f520f4034e1cb06522644c4866056ed0ecce98a2f65706d",
    transactionIndex: 3
}]

> eth.filter({'fromBlock': 'earliest', 'toBlock': 'latest', address: '0xe25b529f5b2475694eeb29dced46dde986488b03', topics: ["0x8a314ef48dbd2506be5878ac08beae6e006dc0e15052df82217252ba71026059", "0x74696d656c6f7264000000000000000000000000000000000000000000000000", null]}).get()
[]

```

One thing I noticed is that this feature seems to work normally on very recent logs (from blocks I mined myself? something to do with syncing?), but not on older ones.

#### Steps to reproduce the behaviour

I'm working on a private chain, in which all nodes are running geth 1.7.0.

My contract is: 

```
contract MassRecord {
        
    event LogRecord (uint a, uint b, bytes32 indexed c, bytes32 indexed d, uint timestamp);
    
    function enter (uint[] a, uint[] b, bytes32[] c, bytes32[] d, uint[] timestamp) {
        for (uint i = 0; i < appid.length; i++) {
            LogRecord(a[i], b[i], c[i], d[i], timestamp[i]);
        }
        
    }        
}

```

#### Backtrace

No backtrace.

**Comment by karalabe** [comment]
Thanks for this. We've also noticed after release. There's a PR open to address this issue https://github.com/ethereum/go-ethereum/pull/15195, will be fixed in Geth 1.7.1 to be released hopefully this week.

**Comment by holiman** [comment]
The PR is merged, I'll close this. Please reopen if the problem remains.
